### PR TITLE
Lowercase license files

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1552)

### CODE REVIEW

File: spdx-utils/src/main/kotlin/SpdxUtils.kt [hunk 1]
```diff
@@ -26,18 +26,26 @@ import java.util.EnumSet
 
 import org.apache.commons.codec.binary.Hex
 import org.apache.commons.codec.digest.DigestUtils
+import java.nio.file.Paths
```
Comment (fviernau) [inline spdx-utils/src/main/kotlin/SpdxUtils.kt]: unrelated.

Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Fixed.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt [hunk 2]
```diff
@@ -32,12 +32,19 @@ import org.apache.commons.codec.digest.DigestUtils
  */
 val LICENSE_FILE_NAMES = listOf(
     "LICENSE*",
+    "license*",
```
Comment (sschuberth) [inline spdx-utils/src/main/kotlin/SpdxUtils.kt]: For me, the commit message lacks a rationale *why* we'd want to support lower-case license file names. By convention, these files should indeed be upper-case.

Also, instead of hard-coding the lower-case variants here, it probably makes more sense to programmatically generate them, e.g. as part of initializing `LICENSE_FILE_MATCHERS`.

File: scanner/src/main/kotlin/scanners/ScanCode.kt [hunk 3]
```diff
@@ -385,13 +385,14 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
         val rootLicenseFile = result["files"].singleOrNull { file ->
             val path = file["path"].textValue()
             try {
-                LICENSE_FILE_MATCHERS.any { it.matches(Paths.get(path)) }
+                file["licenses"]?.singleOrNull() != null
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/ScanCode.kt]: While at it, instead of picking an arbitrary non-empty license, shouldn't we return a license expression with all licenses concatenated here?

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 2 above)
Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: I have added a rationale to the commit mesage. I prefer to keep the plain list for simplicity, it's not a long list. Also, only using lowercase for the matchers could cause inconsistencies.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 3 above)
Comment (mnonnenmacher) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: I agree, but this is a bigger change I did not want to make now.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Commit message typo in "convetion".

I've also seen projects that use a file named "License". Shouldn't we then also support that, and do a fully case-insensitive (glob) match on all platforms? Though I'm not yet sure how to elegantly implement that.

Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Unfortunately globs do not support a case insensitive mode. I think it would be best to just add "License" and similar to the list.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 3 above)
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Well, ok, but could you please add a `TODO` here so we don't forget about this when starting to use license expressions throughout this file?

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Now that we basically agreed that we need three variants, upper-case, lower-case and capitalized, I'd like to again propose a programmatic approach, this time directly when initializing `LICENSE_FILE_NAMES`:

```
val LICENSE_FILE_NAMES = listOf(
    "license*",
    "licence*",
    "unlicense",
    "unlicence",
    "copying*",
    "copyright",
    "patents"
).flatMap { listOf(it, it.toUpperCase(), it.capitalize()) }
```

Note that the list needs to contain the lower-case strings now, or else `capitalize()` would not work.

Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Now that we basically agreed that we need three variants, upper-case, lower-case and capitalized, I'd like to again propose a programmatic approach, this time directly when initializing `LICENSE_FILE_NAMES`:

```
val LICENSE_FILE_NAMES = listOf(
    "license*",
    "licence*",
    "unlicense",
    "unlicence",
    "copying*",
    "copyright",
    "patents"
).flatMap { listOf(it, it.toUpperCase(), it.capitalize()) }
```

Note that the list needs to contain the lower-case strings now, or else `capitalize()` would not work.

Comment (mnonnenmacher) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Done.

File: scanner/src/main/kotlin/scanners/ScanCode.kt (hunk 3 above)
Comment (mnonnenmacher) [reply scanner/src/main/kotlin/scanners/ScanCode.kt]: Done.
